django_netjsonconfig/api/generics.py

```diff
@@ -1,7 +1,21 @@
-from rest_framework.generics import ListAPIView
+from rest_framework.generics import RetrieveAPIView, ListAPIView, get_object_or_404
 from rest_framework.response import Response
 
-from .serializers import SearchTemplateSerializer
+from .serializers import TemplateRetrieveSerializer, SearchTemplateSerializer
+
+
+class BaseGetTemplateView(RetrieveAPIView):
+    serializer_class = TemplateRetrieveSerializer
+
+    def get(self, request, *args, **kwargs):
+        key = request.GET.get('key', None)
+        if key:
+            temp = get_object_or_404(self.template_model, pk=kwargs['uuid'], key=key,
+                                     sharing='secret_key')
+        else:
+            temp = get_object_or_404(self.template_model, pk=kwargs['uuid'], sharing='public')
```

Comment:
let's avoid repeating `get_object_or_404`, the parameters are mostly identical. You can use something like the following:

```python
opts = {'pk': kwargs['uuid'], 'sharing': 'public'}
if key:
    opts.update({
        'key': key,
        'sharing': 'secret-key'
    })
template = get_object_or_404(self.template_model, **opts)
```